subprojects/plugin-development/src/integTest/groovy/org/gradle/plugin/devel/tasks/ValidatePluginsIntegrationTest.groovy

```diff
@@ -571,14 +565,66 @@ class ValidatePluginsIntegrationTest extends AbstractPluginValidationIntegration
-            error(unsupportedValueType { type('MyTask').property('setPropertyInput').annotationType(annotation).unsupportedValueType('ResolvedArtifactResult').propertyType('SetProperty<ResolvedArtifactResult>').solution('Extract artifact metadata and annotate with @Input.').solution('Extract artifact files and annotate with @InputFiles.') }, "validation_problems", "unsupported_value_type"),
+        assertValidationFailsWith([ // Note that only the first WorkValidationException.MAX_ERR_COUNT of these will be reported, so don't test for more than that
+                                    error(unsupportedValueType { type('MyTask').property('direct').annotationType(annotation).unsupportedValueType('ResolvedArtifactResult').propertyType('ResolvedArtifactResult').solution('Extract artifact metadata and annotate with @Input.').solution('Extract artifact files and annotate with @InputFiles.') }, "validation_problems", "unsupported_value_type"),
+                                    error(unsupportedValueType { type('MyTask').property('listPropertyInput').annotationType(annotation).unsupportedValueType('ResolvedArtifactResult').propertyType('ListProperty<ResolvedArtifactResult>').solution('Extract artifact metadata and annotate with @Input.').solution('Extract artifact files and annotate with @InputFiles.') }, "validation_problems", "unsupported_value_type"),
+                                    error(unsupportedValueType { type('MyTask').property('mapPropertyInput').annotationType(annotation).unsupportedValueType('ResolvedArtifactResult').propertyType('MapProperty<String, ResolvedArtifactResult>').solution('Extract artifact metadata and annotate with @Input.').solution('Extract artifact files and annotate with @InputFiles.') }, "validation_problems", "unsupported_value_type"),
+                                    error(unsupportedValueType { type('MyTask').property('nestedBean.nestedInput').annotationType(annotation).unsupportedValueType('ResolvedArtifactResult').propertyType('Property<ResolvedArtifactResult>').solution('Extract artifact metadata and annotate with @Input.').solution('Extract artifact files and annotate with @InputFiles.') }, "validation_problems", "unsupported_value_type"),
+                                    error(unsupportedValueType { type('MyTask').property('propertyInput').annotationType(annotation).unsupportedValueType('ResolvedArtifactResult').propertyType('Property<ResolvedArtifactResult>').solution('Extract artifact metadata and annotate with @Input.').solution('Extract artifact files and annotate with @InputFiles.') }, "validation_problems", "unsupported_value_type")
+        ])
+
+
+        where:
+        annotation   | _
+        "Input"      | _
+        "InputFile"  | _
+        "InputFiles" | _
+    }
+
+    @ValidationTestFor(ValidationProblemId.UNSUPPORTED_VALUE_TYPE)
+    def "can not use ResolvedArtifactResult as task input annotated with @#annotation - part 2"() {
```

Comment:
There's notes in the `expect` block: _only the first WorkValidationException.MAX_ERR_COUNT of these will be reported, so don't test for more than that_.  So I broke up the test method so that the number of errors in the test in excess of `MAX_ERR_COUNT` were tested in a new method.  There wasn't really a good seam to divide the 8 errors along that I could see, so I figured part 2 would at least make people curious to check both methods when changing either.

An alternate would be to add a non-final static field on `WorkValidatedException` to hold the current error count, and initialize it from the default `MAX_ERR_COUNT`, but allow the tests to update it to a higher count.  Want to do this?

Reply:
Nevermind all that, I somehow missed until now that we're reading a system property that we can just set to a higher value when working with these tests.  I'll adjust it and merge these tests.